src/cpp/cpp_typecheck_conversions.cpp

```diff
@@ -607,7 +607,7 @@ bool cpp_typecheckt::standard_conversion_pointer_to_member(
     to_struct_type(follow(static_cast<const typet &>
       (type.find("to-member"))));
 
-  if(subtype_typecast(to_struct, from_struct))
+  if(subtype_typecast(from_struct, to_struct))
```

Comment:
I believe this change would be wrong: this method is about downcasting member pointers; this test checks whether it genuinely is a downcast.